pkg/datapath/loader/tc_test.go

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package loader
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+
+	"github.com/cilium/cilium/pkg/testutils"
+	"github.com/cilium/cilium/pkg/testutils/netns"
+)
+
+func newTCProgram() (*ebpf.Program, error) {
+	return ebpf.NewProgram(&ebpf.ProgramSpec{
+		Type: ebpf.SchedCLS,
+		Instructions: asm.Instructions{
+			asm.Mov.Imm(asm.R0, 0),
+			asm.Return(),
+		},
+		License: "Apache-2.0",
+	})
+}
+
+func mustTCProgram(tb testing.TB) *ebpf.Program {
+	p, err := newTCProgram()
+	if err != nil {
+		tb.Skipf("tc programs not supported: %s", err)
+	}
+	tb.Cleanup(func() {
+		p.Close()
+	})
+	return p
+}
+
+func mustTCProgramWithName(t *testing.T, name string) *ebpf.Program {
+	p, err := ebpf.NewProgram(&ebpf.ProgramSpec{
+		Name: name,
+		Type: ebpf.SchedCLS,
+		Instructions: asm.Instructions{
+			asm.Mov.Imm(asm.R0, 0),
+			asm.Return(),
+		},
+		License: "Apache-2.0",
+	})
+	if err != nil {
+		t.Skipf("tc programs not supported: %s", err)
+	}
+	t.Cleanup(func() {
+		p.Close()
+	})
+	return p
+}
+
+func TestAttachDetachSKBProgram(t *testing.T) {
```

Comment:
Should this test be run with tcx disabled? We already cover all the TCX cases with other tests, so this one should maybe focus on legacy tc attach/detach.

Reply:
Good point, I think I had this in mind but it fell through. I ended up renaming this to `TestAttachDetachSKBProgramLegacy` and adding a separate `TestAttachDetachTCProgram` that only uses the legacy tc APIs. :+1: